### DISCUSSION TITLE
Cart Button Does Not Appear At All

### DISCUSSION THREAD

**Discussion by GP-Legend** [solidusio/solidus#4668] [Troubleshooting] [answered]
The cart button does not appear in any way on a single page of the entire application for some reason. This has occurred on a fresh install of the Solidus gem and I have not made a single modification of any kind to the code. The source of the bug is an undefined 'Solidus' reference, which I have included a picture of.

**Solidus Version**:
This is happening on version 3.1.8.

**To Reproduce**
Install Solidus inside of a Rails application via the instructions on the application's main GitHub page. Choose the solidus_starter_frontend from the options menu when it appears. After the installation is finished, simply open up the application on a local server and observe how the cart button is missing, thus making it impossible for potential users to finalize any sort of purchases.

**Current behavior**
As was explained before, the cart button is not present at all, negating the primary function of an e-commerce application.

**Expected behavior**
The cart button should be present.

**Screenshots**

![Cart Button Missing](https://user-images.githubusercontent.com/113549664/194614423-6246706e-7f4c-4679-ab2c-b871d027fd80.png)
![Undefined Solidus](https://user-images.githubusercontent.com/113549664/194614451-fd930a0d-f95b-4f1a-8be5-82b5545566af.png)

**Desktop (please complete the following information):**
 - OS: Mac OS Big Sur
 - Browser: Google Chrome
 - Version: Version 105.0.5195.125 (Official Build) (x86_64)

**Additional context**
This glaring bug was found after an attempt to follow Rails programmer Deanin's YouTube tutorial on how to set up the store (https://www.youtube.com/watch?v=Bd4y64Dk5QU). In his video you can see that the cart button is present, but if you try and follow the same steps he did, the button is now gone, and it is not addressed in the video.

I suspect that this is due to the fact that the video was recorded before and then released on September 12 of this year, but the current version I installed was released fifteen days ago on September 22. Perhaps that was enough time for a development to be made which broke one of the features of the application? In any case, how can I go about fixing this problem? I am about to begin styling the store for myself using the official guide, and any help at all is greatly appreciated. Thank you!

**Comment by jarednorman** [comment]
The fact that you're getting both errors around jQuery and Spree not being loaded, suggests none of the JS is being loaded.

What version of Rails are you using and are you just using Sprockets or one of the other JS options now available?

**Comment by GP-Legend** [comment]
> The fact that you're getting both errors around jQuery and Spree not being loaded, suggests none of the JS is being loaded.
> 
> What version of Rails are you using and are you just using Sprockets or one of the other JS options now available?

I am currently using Rails 7.0.4. I don't have any knowledge about Sprockets or anything like that, is it some sort of Javascript framework you need in order for Rails and Javascript to cooperate?

**Comment by jarednorman** [comment]
Sprockets is the default frontend asset management pipeline in Rails. What was the command you used to generate your Rails application?

**Comment by GP-Legend** [comment]
> Sprockets is the default frontend asset management pipeline in Rails. What was the command you used to generate your Rails application?

I typed in 'rails new solid2'* and then installed the Solidus gem.

*solid2 just stands for 'Solidus 2' as this is my second time trying to set up an application with it.

**Comment by elia** [comment, marked as answer]
I tried to recreate the issue locally, I was able to make it work with the latest solidus and rails:

```
rails new solid2
cd solid2
bundle add solidus
bin/rails g solidus:install # taking all the default options
bin/rails s # the cart button shows up, no errors in console
```

but I had an issue trying to make it work with Rails 7.0.4 and Solidus 3.1.8, as it can only be installed on rails < 6.2:

```
Bundler could not find compatible versions for gem "railties":
  In Gemfile:
    rails (~> 7.0.4) was resolved to 7.0.4, which depends on
      railties (= 7.0.4)

    solidus (= 3.1.8) was resolved to 3.1.8, which depends on
      solidus_core (= 3.1.8) was resolved to 3.1.8, which depends on
        railties (>= 5.2, < 6.2.x)
```

_(I moved this issue to a troubleshooting discussion so the resolving answer can be marked and help other users)_

**Reply by GP-Legend** [reply]
It worked, so grazie mille for your post! However, I suspect now that the issue lies in when I installed Solidus Stripe. I will test it right now and report back. If it causes any problems, I will mark what you said as the answer and try to warn others not to install solidus_stripe until some sort of fix is available.

**Comment by GP-Legend** [comment]
Yep, it looks like installing Solidus Stripe causes the gem's Javascript to malfunction for some reason. Looks like I'll have to start my project over again, but that's okay.

Thank you @elia and @jarednorman for your help! I appreciate it a lot!

**Reply by elia** [reply]
solidus_stripe and the other payment extensions are not yet fully compatible with solidus_starter_frontend, solidus 2.3 should have proper warnings and options in the installer for bolt and paypal (see https://github.com/solidusio/solidus/pull/4659).

Restoring compatibility is in the roadmap tho, so it should be just a matter of time.

**Reply by GP-Legend** [reply]
Take as long as you guys need to! This gem is already quite helpful even as it already is.
